assignments/this.js

```diff
@@ -1,26 +1,49 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. The first rule of this is when a function is available in the global scope, the value of this 
+    becomes the window object.
+
+* 2. The second rule of this is when a function is called by a preceding dot, this becomes the object before the 
+    preceding dot.
+
+
+* 3. The third principle of this key word is usage of `new` keyword when instantiating a constructor function.
+    this becomes the instance of the function.
+
+* 4. The fourth rule of this is using call(), or apply(), or bind() to explicitly bind this.
 *
 * write out a code example of each explanation above
 */
+function GameObject(name, dimensions, createdAt) {
+    this.name = name;
+    this.createdAt = createdAt;
+    this.dimensions = dimensions;
+  }
+  GameObject.prototype.destroy = function(game) {
+    return `${this.name} was removed from the ${game} game.`;
+  }
 
 // Principle 1
-
 // code example for Window Binding
+const gameObjectWindowBinding = function() {
+    console.log(this);
+}
+console.log(gameObjectWindowBinding());
 
 // Principle 2
-
 // code example for Implicit Binding
+  const gameObjectImplicitBinding = new GameObject('Ezekiel',   { length: 2, width: 1, height: 1},  new Date());
```

Comment:
Be careful here, you are using the new Keyword here, and indeed the `this` is referred to the newly created object. 
`const giacomo = {name: "Giacomo", greet() {return "hi I'm ${this.name} }}`
Notice how here `this is referred to the object itself.